Source/JSQSystemSoundPlayer.m

```diff
@@ -122,39 +120,30 @@ - (void)setBundle:(NSBundle *)bundle
 
 #pragma mark - Playing sounds
 
-- (void)playSoundWithName:(NSString *)filename
-                extension:(NSString *)extension
-                  isAlert:(BOOL)isAlert
-          completionBlock:(JSQSystemSoundPlayerCompletionBlock)completionBlock
+- (void)playSoundWithSoundID:(SystemSoundID)soundID
+                     isAlert:(BOOL)isAlert
+             completionBlock:(JSQSystemSoundPlayerCompletionBlock)completionBlock
 {
-    NSParameterAssert(filename != nil);
-    NSParameterAssert(extension != nil);
-
     if (!self.on) {
         return;
     }
 
-    if (![self.sounds objectForKey:filename]) {
-        [self addSoundIDForAudioFileWithName:filename extension:extension];
-    }
-
-    SystemSoundID soundID = [self soundIDForFilename:filename];
```

Comment:
the re-ordering here makes this diff a bit hard to read. let's leave everything above this line how it was, then extract what's below this line into the new method, `playSoundForSoundID: asAlert: completion:`